### PR TITLE
lndclient: add graph topology change subscription

### PR DESCRIPTION
**Pull request by bhandras**

### CODE REVIEW

File: lightning_client.go
Comment (guggero) [inline lightning_client.go:1954]: :+1:

File: lightning_client.go
Comment (guggero) [inline lightning_client.go:2867]: nit: add a comment that this is always set in `lnd`'s `rpcserver.go` and therefore should never run into a nil reference error.

**Review by guggero — APPROVED** [review]
LGTM, nice addition :100:

**Comment by bhandras** [comment]
Since toplology change notifications are changing (will use feature map instead of serialized features) I'm putting this PR on hold.
More context: https://github.com/lightningnetwork/lnd/pull/4822

File: lightning_client.go
Comment (bhandras) [reply lightning_client.go:2867]: Sorry, I pushed the merge button in the GitHub app for the wrong PR, will create a follow up (also making support for the new `Features` map).

**Pull request merged**
